src/definitions/device_engagement.rs

```diff
@@ -21,69 +27,100 @@ pub type ProtocolInfo = CborValue;
 pub type Oidc = (u64, String, String);
 pub type WebApi = (u64, String, String);
 
+/// Represents a device engagement.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 #[serde(try_from = "CborValue", into = "CborValue", rename_all = "camelCase")]
 pub struct DeviceEngagement {
+    /// The version of the device engagement.
     pub version: String,
+
+    /// The security settings for the device engagement.
     pub security: Security,
-    #[serde(skip_serializing_if = "Option::is_none")]
+
+    /// The optional device retrieval methods for the device engagement.
     pub device_retrieval_methods: Option<DeviceRetrievalMethods>,
-    #[serde(skip_serializing_if = "Option::is_none")]
+
+    /// The optional server retrieval methods for the device engagement.
     pub server_retrieval_methods: Option<ServerRetrievalMethods>,
-    #[serde(skip_serializing_if = "Option::is_none")]
+
+    /// The optional protocol information for the device engagement.
     pub protocol_info: Option<ProtocolInfo>,
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 #[serde(try_from = "CborValue", into = "CborValue")]
 pub enum DeviceRetrievalMethod {
+    /// Represents the options for a WiFi connection.
     WIFI(WifiOptions),
+
+    /// Represents the BLE options for device engagement.
+    ///
+    /// This struct is used to configure the BLE options for device engagement.  
+    /// It contains the necessary parameters and settings for BLE communication.
     BLE(BleOptions),
+
+    /// Represents the options for NFC engagement.
     NFC(NfcOptions),
 }
 
+/// Represents the bytes of an EDevice key.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct Security(pub u64, pub EDeviceKeyBytes);
 
+/// Represents the server retrieval methods for device engagement.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 #[serde(rename_all = "camelCase")]
 pub struct ServerRetrievalMethods {
-    #[serde(skip_serializing_if = "Option::is_none")]
+    /// The `web API retrieval method. This field is optional and will be skipped during serialization if it is [None].
     web_api: Option<WebApi>,
-    #[serde(skip_serializing_if = "Option::is_none")]
+
+    /// The `OIDC`` retrieval method. This field is optional and will be skipped during serialization if it is [None].
     oidc: Option<Oidc>,
 }
```

Comment:
Was it intended to remove `#[serde(skip_serializing_if = "Option::is_none")]` for these fields? If so, we may wish to  adjust the comments.

An a related note, there may be a prior discussion around this that I am unfamiliar with, but should we use an enum here instead to represent that at least one of those optional fields should be required? I don't think that is required for this PR, but one we may wish to consider in the future.

Reply:
On further review of the specification, it would be valid for both methods to be transferred as part of the device engagement. So, it makes sense that we use a struct here instead of an enum.

Reply:
No, copilot mistakes which seems I missed. Thank you for noticing.